### PR TITLE
fix: custom HTML color

### PR DESCRIPTION
**Pull request by BisuGit**
## Description
added `replaceColor` function

### DevQA Prep
- add this to the examples `<span style="color:rgb(33, 102, 216);">“The 9/11 Commission Report, completed in 2004 by a team appointed by President Bush and members of Congress, recommended a series of changes in government policy to respond to the threat of modern terrorism.”&nbsp;</span>`
- Run `pod install`
- Test `RichTextView` to ensure that it passes (unit and UI tests)
- Navigate to the `Example` project in the root `Example` folder and run the app
- Ensure that everything in the `Example` app looks visually correct

### DevQA Steps
run the example project and check if the string is blue.

### CODE REVIEW

**Comment by tophat-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Please ensure that your new files respect the existing folder structure</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">
<p>If anything other than <code>RichTextView</code> appears in the code coverage section please add it<br />
to the <code>exclude_targets</code> property of the <code>xcov.report()</code> function in the Dangerfile</p>
</td>
    </tr>
  </tbody>
</table>

## Current coverage for RichTextView.framework is `90.43%`
Files changed | - | - 
--- | --- | ---
UIColorExtension.swift | `72.73%` | :warning:
RichTextView.swift | `86.86%` | :white_check_mark:
RichTextParser.swift | `94.62%` | :white_check_mark:
UITextViewGenerator.swift | `95.00%` | :white_check_mark:
NSMutableAttributedStringExtension.swift | `100.00%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Source/Extensions/NSMutableAttributedStringExtension.swift [hunk 1]
```diff
@@ -23,6 +23,17 @@ extension NSMutableAttributedString {
         self.endEditing()
     }
 
+    func replaceColor(with newColor: UIColor) {
+        self.beginEditing()
+//        let defaultColor = UIColor.black
```
Comment (aelkady) [inline Source/Extensions/NSMutableAttributedStringExtension.swift]: I would keep this defaultColor and set it to black - with the UIColor extension the equality comparison should still work. Also this method needs tests.

File: Source/Extensions/UIColorExtension.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+//  UIColorExtension.swift
+//  RichTextView
+//
+//  Created by Simon Chau on 2019-07-03.
+//  Copyright © 2019 Top Hat. All rights reserved.
+//
+
+extension UIColor {
+    static func == (leftColor: UIColor, rightColor: UIColor) -> Bool {
+        var red1: CGFloat = 0
```
Comment (aelkady) [inline Source/Extensions/UIColorExtension.swift]: Can we rename `red1` to `leftRed`, `red2` to `rightRed` and the same for the rest of the colors.

Also this function needs tests.

File: UITests/RichTextViewUITests.swift [hunk 3]
```diff
@@ -30,112 +30,112 @@ class RichTextViewUITests: QuickSpec {
...
-                    self.viewController?.view.addSubview(richTextView)
-                    self.window?.rootViewController = self.viewController
-                    waitUntil(timeout: Defaults.timeOut) { done in
-                        DispatchQueue.main.asyncAfter(deadline: .now() +  Defaults.delay, execute: {
-                            expect(self.window).to(haveValidSnapshot())
-                            done()
-                        })
-                    }
-                }
+//                it("Renders a regular string with no rich text formatting") {
```
Comment (aelkady) [inline UITests/RichTextViewUITests.swift]: Uncomment these tests.

File: UnitTests/Extensions/NSMutableAttributedStringExtensionSpec.swift [hunk 4]
```diff
@@ -14,6 +14,18 @@ import Nimble
 class NSMutableAttributedStringExtensionSpec: QuickSpec {
     override func spec() {
         describe("NSMutableAttributedString Extension") {
+            context("Replace Font Color") {
+                it("successfully replaces the font color") {
+                    let attributedText = NSMutableAttributedString(string: "Hello, world", attributes: [
+                        .foregroundColor: UIColor.black
+                        ])
```
Comment (aelkady) [inline UnitTests/Extensions/NSMutableAttributedStringExtensionSpec.swift]: Indent fix. Also each parameter on a different line.

**Pull request merged**